src/Data/ArrayBuffer/ArrayBuffer.purs

```diff
@@ -1,52 +1,31 @@
-module Data.ArrayBuffer.ArrayBuffer ( create
-                                    , byteLength
-                                    , slice
-                                    , fromArray
-                                    , fromIntArray
-                                    , fromString
-                                    , decodeToString
-                                   ) where
+-- | This module represents the functional bindings to JavaScript's `ArrayBuffer`
+-- | objects. See [MDN's spec](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/ArrayBuffer) for details.
+
+module Data.ArrayBuffer.ArrayBuffer
+  ( empty
+  , byteLength
+  , slice
+  ) where
 
-import Data.ArrayBuffer.DataView (whole, buffer)
-import Data.ArrayBuffer.Typed (asUint8Array, dataView)
 import Data.ArrayBuffer.Types (ArrayBuffer, ByteOffset, ByteLength)
-import Data.Either (Either)
 import Data.Function.Uncurried (Fn3, runFn3)
-import Data.TextDecoder (decodeUtf8)
-import Data.TextEncoder (encodeUtf8)
-import Effect (Effect)
-import Effect.Exception (Error)
-import Prelude ((<<<))
+import Data.Maybe (Maybe(..))
+import Data.Nullable (Nullable, notNull, null)
+import Data.Tuple (Tuple(..))
 
 
 -- | Create an `ArrayBuffer` with the given capacity.
-foreign import create :: ByteLength -> Effect ArrayBuffer
+foreign import empty :: ByteLength -> ArrayBuffer
```

Comment:
Do you think this should be an `Effect`?

Reply:
I think it should, since:
```
let a = empty 16
foo a
bar a
```

Is quite different from

```
foo $ empty 16
bar $ empty 16
```

right?

Reply:
Yes, I guess the right thing would be to make `toArray` effectful.

Oh, I switched right now to the v10 branch. I changed there the order of arguments for the `DataView` setters and the implementation. I don't like the `ViewTypeName` stuff there, maybe you can figure out some type-level black magic to avoid it?

Reply:
Awesome, I like the new DataView. I'm not sure if I can get away with any typelevel magick here - purescript doesn't really have a type-level `class Show :: Type -> Symbol -> Constraint` typeclass in typelevel-prelude, so there's not really an automated way of printing out typenames. If there's a sneaky way of doing this, I'm all ears. But I think that might just have to wait for the next version of purs or something :\. The least I could do right now is omit the AProxy stuff from this typeclass - we'll just make it a type-level mapping, then reflect the type back to a string.

Reply:
Nice!